.github/workflows/lintdiff.yaml

```diff
@@ -0,0 +1,68 @@
+name: LintDiff (preview)
+
+on: pull_request
+
+permissions:
+  contents: read
+
+jobs:
+  lintdiff:
+    name: LintDiff (preview)
+    runs-on: ubuntu-24.04
+
+    steps:
+      - name: Checkout eng
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            eng/
+            .github/
+
+      - name: Checkout 'after' state
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+          path: after
+
+      - name: Checkout 'before' state
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.base.sha }}
+          path: before
+
+      - name: Setup Node and install deps
+        uses: ./.github/actions/setup-node-install-deps
+        with:
+          working-directory: eng/tools/lint-diff
+          # TODO: this probably needs a package-lock.json and dist
+          install-command: npm i
+      - run: npm run build
+        working-directory: eng/tools/lint-diff
+
+      # TODO: This can probably be updated to run directly from JS
+      # TODO: default workspace is the after/ folder
+      - name: Get changed files
```

Comment:
I will be creating JS-based helpers for this: https://github.com/Azure/azure-rest-api-specs/issues/32637

Reply:
Helper is ready: https://github.com/Azure/azure-rest-api-specs/blob/main/.github/src/changed-files.js

Reply:
Started using JS helpers. More work to do here.

Reply:
https://github.com/Azure/azure-rest-api-specs/issues/33302